internal/functions/coviddata/download-covid-data-total.go

```diff
@@ -137,9 +137,10 @@ func DownloadCovidDataTotal(w http.ResponseWriter, r *http.Request) {
 				return fmt.Errorf("Error while querying Firestore: %v", err)
 			}
 
-			return tx.Set(doc, *d)
+			return tx.Set(doc, *totalsData)
 		}
-		return nil
+
+		return tx.Set(doc, *totalsData)
```

Comment:
are you sure this will work? wouldn't the first return fmt.Errorf be returned first without ever reaching the tx.Set? Since the functionality is changing, I think it is reasonable to simplify this too and possibly get rid of the fmt.Errorf (maybe change it to Debugf("we didn't find a key and we are creating a new record") vs. Debugf("mutating data for this date"). wdyt?

Reply:
There are three cases:
1. The query fails because of error.
1. The query fails because the record doesn't exist yet.
1. The query succeeds.

`return fmt.Errorf` is only for the first case. In the second, data are just stored. In the third, the data can be thrown away and be overwritten.  
You are, however, right, that this can be simplified - the transaction is now not needed at all. Updated.